### PR TITLE
Module update to new paradigm with backwards compatibility

### PR DESCRIPTION
**Pull request by agenius-mohammed-ali**

### CODE REVIEW

File: R/appReset.R [hunk 1]
```diff
@@ -25,21 +25,60 @@
...
+    
+    if (old_style_call) {
+        app_reset(input, output, session, logger)
+    }
+    else {
+        moduleServer(
+            id,
+            function(input, output, session) {
+                app_reset(input, output, session, logger)
+
```
Comment (cb4ds) [inline R/appReset.R:61]: formatting

File: R/appReset.R [hunk 2]
```diff
@@ -25,21 +25,60 @@
...
 # Module Server Function
-.appReset <- function(input, output, session, logger) {
+.appReset <- function(..., logger) {
+    call <- match.call()
+    params <- list(...)
+    param_index <- 1
+    params_length <- length(params)
+    old_style_call <- call[[1]] == "module" || "periscope" %in% as.character(call[[1]])
+    
+    # get session parameters
```
Comment (cb4ds) [inline R/appReset.R:35]: remove comment

File: R/appReset.R [hunk 3]
```diff
@@ -25,21 +25,60 @@
...
+        output  <- params[[param_index]]
+        param_index <- param_index + 1
+        session <- params[[param_index]]
+        param_index <- param_index + 1
+    } else {
+        id <- params[[param_index]]
+        param_index <- param_index + 1
+    }
+    
+    # get rest of the function parameters
```
Comment (cb4ds) [inline R/appReset.R:48]: remove comment

File: R/bodyFooter.R [hunk 4]
```diff
@@ -19,27 +19,64 @@
...
-.bodyFooter <- function(input, output, session, logdata) {
-    output$dt_userlog <- shiny::renderTable({
+.bodyFooter <- function(..., logdata) {
+    call <- match.call()
+    params <- list(...)
+    param_index <- 1
+    params_length <- length(params)
+    old_style_call <- call[[1]] == "module" || "periscope" %in% as.character(call[[1]])
+    
+    # get session parameters
```
Comment (cb4ds) [inline R/bodyFooter.R:29]: remove comment

File: R/bodyFooter.R [hunk 5]
```diff
@@ -19,27 +19,64 @@
...
+        output  <- params[[param_index]]
+        param_index <- param_index + 1
+        session <- params[[param_index]]
+        param_index <- param_index + 1
+    } else {
+        id <- params[[param_index]]
+        param_index <- param_index + 1
+    }
+    
+    # get rest of the function parameters
```
Comment (cb4ds) [inline R/bodyFooter.R:42]: remove comment

**Review by cb4ds — CHANGES_REQUESTED** [review]
- Update the version to 1.0.0.900x
- Resolve the 2 warnings and 1 note in the R CMD check
- Run devtools::spell_check (and fix the one odd spelling item noted below) and double check changes to vignettes/etc.'
- Double check formatting changes
- There are some comments that aren't necessary - I flagged a few, please review and follow through the rest of the files for those specifically.  Since you are doing that over and over pull the logic out into a helper function that is internal, that's far better documentation.

File: R/bodyFooter.R [hunk 6]
```diff
@@ -19,27 +19,64 @@
...
+    else {
+        moduleServer(
+            id,
+            function(input, output, session) {
+                boody_footer(input, output, session, logdata)
+            })
+    }
+}
 
+boody_footer <- function(input, output, session, logdata) {
```
[the remark this answers is not in the document]
Comment (agenius-mohammed-ali) [reply R/bodyFooter.R:59]: done

File: R/appReset.R (hunk 1 above)
Comment (agenius-mohammed-ali) [reply R/appReset.R:61]: done

File: R/appReset.R (hunk 2 above)
Comment (agenius-mohammed-ali) [reply R/appReset.R:35]: done

File: R/appReset.R (hunk 3 above)
Comment (agenius-mohammed-ali) [reply R/appReset.R:48]: done

File: R/bodyFooter.R (hunk 5 above)
Comment (agenius-mohammed-ali) [reply R/bodyFooter.R:42]: done

File: R/bodyFooter.R (hunk 4 above)
Comment (agenius-mohammed-ali) [reply R/bodyFooter.R:29]: - done, for the comment part.
- for extracting the logic into a separate function, that is what I want to do first time. But for some reason, seasion parameters got corrupted if I used them in a do.call if I build the functions arguments dynamically or passed them as a list to use them. So I had to let each function build its own arguments using the passed session parameter as they are with no change or indirect calls. Thank you

**Review by cb4ds — APPROVED** [review]
Approved as WIP, thanks!

**Review by cb4ds — APPROVED** [review]

**Pull request merged**